openmetadata-ui/src/main/resources/ui/src/components/common/EntityPageInfos/AnnouncementDrawer/AnnouncementDrawer.tsx

```diff
@@ -64,92 +61,96 @@ const AnnouncementDrawer: FC<Props> = ({
-    try {
-      await postThread(data);
-    } catch (err) {
-      showErrorToast(err as AxiosError);
-    }
-  };
-
-  const deletePostHandler = (
+  const deletePostHandler = async (
     threadId: string,
     postId: string,
     isThread: boolean
-  ) => {
-    deletePost(threadId, postId, isThread);
+  ): Promise<void> => {
+    await deletePost(threadId, postId, isThread);
   };
 
-  const postFeedHandler = (value: string, id: string) => {
+  const postFeedHandler = async (value: string, id: string): Promise<void> => {
     const data = {
       message: value,
       from: currentUser?.name,
     } as Post;
-    postFeedById(id, data).catch((err: AxiosError) => {
-      showErrorToast(err);
-    });
+
+    try {
+      await postFeedById(id, data);
+    } catch (err) {
+      showErrorToast(err as AxiosError);
+    }
   };
 
-  const updateThreadHandler = (
+  const updateThreadHandler = async (
     threadId: string,
     postId: string,
     isThread: boolean,
     data: Operation[]
-  ) => {
+  ): Promise<void> => {
     const callback = () => {
       return;
     };
 
-    updateThreadData(threadId, postId, isThread, data, callback);
+    await updateThreadData(threadId, postId, isThread, data, callback);
   };
 
+  const handleCloseAnnouncementModal = useCallback(
+    () => setIsAddAnnouncementOpen(false),
+    []
+  );
+  const handleOpenAnnouncementModal = useCallback(
+    () => setIsAddAnnouncementOpen(true),
+    []
+  );
+
+  const handleSaveAnnouncement = useCallback(() => {
+    handleCloseAnnouncementModal();
+    setRefetchThread((prev) => !prev);
+  }, []);
+
   return (
-    <>
-      <div data-testid="announcement-drawer">
-        <Drawer
-          closable={false}
-          open={open}
-          placement="right"
-          title={title}
-          width={576}
-          onClose={onClose}>
-          <div className="d-flex justify-end">
-            <Tooltip
-              title={!createPermission && t('message.no-permission-to-view')}>
-              <Button
-                data-testid="add-announcement"
-                disabled={!createPermission}
-                type="primary"
-                onClick={() => setIsAnnouncement(true)}>
-                {t('label.add-entity', { entity: t('label.announcement') })}
-              </Button>
-            </Tooltip>
-          </div>
-
-          <ActivityThreadPanelBody
-            className="p-0"
-            createThread={createThread}
-            deletePostHandler={deletePostHandler}
-            editAnnouncementPermission={createPermission}
-            key={uniqueId()}
-            postFeedHandler={postFeedHandler}
-            showHeader={false}
-            threadLink={getEntityFeedLink(entityType, entityFQN)}
-            threadType={ThreadType.Announcement}
-            updateThreadHandler={updateThreadHandler}
-          />
-        </Drawer>
-      </div>
-
-      {isAnnouncement && (
+    <div data-testid="announcement-drawer">
```

Comment:
Why we need a div to wrap drawer? can we assign data-testid directly to Drawer instead?